### PR TITLE
Update ButtonBase text variant and fix font smoothing

### PR DESCRIPTION
**Pull request by garrettbear**
## Explanation
- Update `ButtonBase` TextVariant to bodyMdMedium (so it is now medium font weight)
- Update to `ButtonBase` will then automatically update ButtonPrimary, ButtonSeconday, and ButtonLink
- These Buttons are used widely so a significant number of snapshots are required to be updated

* Fixes: #19852 
* 

## Screenshots/Screencaps

```Screenshots attached down below```

## Manual Testing Steps

Compare current [Storybook build](https://metamask.github.io/metamask-storybook/?path=/docs/components-componentlibrary-buttonbase--docs) vs this [PR Storybook](http://localhost:6006/?path=/docs/components-componentlibrary-buttonbase--docs)

## Pre-merge author checklist

- [X] I've clearly explained:
  - [X] What problem this PR is solving
  - [X] How this problem was solved
  - [X] How reviewers can test my changes
- [ ] Sufficient automated test coverage has been added

## Pre-merge reviewer checklist

- [X] Manual testing (e.g. pull and build branch, run in browser, test code being changed)
- [X] PR is linked to the appropriate GitHub issue
- [ ] **IF** this PR fixes a bug in the release milestone, add this PR to the release milestone

If further QA is required (e.g. new feature, complex testing steps, large refactor), add the `Extension QA Board` label.

In this case, a QA Engineer approval will be be required.

### CODE REVIEW

**Review by alokeutpal — APPROVED** [review]

**Review by NidhiKJha — DISMISSED** [review]

File: ui/components/component-library/text/text.scss [hunk 1]
```diff
@@ -32,6 +32,9 @@ $text-variants: (
   // Set default styles
   color: var(--color-text-default);
   font-family: var(--font-family-sans);
+  -webkit-font-smoothing: antialiased;
+  -moz-osx-font-smoothing: grayscale;
+  font-smooth: never;
```
[the remark this answers is not in the document]
Comment (georgewrmarshall) [reply ui/components/component-library/text/text.scss:37]: Do you think turning this on will effect readability? I tend to lean toward bolder, bigger, higher contrast to improve accessibility for vision impaired folks?

[the remark this answers is not in the document]
Comment (georgewrmarshall) [reply ui/components/component-library/text/text.scss:37]: For how subtle the difference I'm not sure we want to use font smoothing. It sounds like setting `font-smooth: never;` will render the font pretty inconsistently across browsers and make the font weight much thinner. I tend to lean toward bolder, bigger, higher contrast to improve accessibility for vision impaired folks? It also mentions not to use it in production on mdn web docs https://developer.mozilla.org/en-US/docs/Web/CSS/font-smooth

Screenshot 2023-07-11 at 3 57 39 PM
Screenshot 2023-07-11 at 3 58 06 PM

**Review by georgewrmarshall — COMMENTED** [review]
Font weight looks great. Not sure about the font smoothing

**Review by georgewrmarshall — APPROVED** [review]
LGTM! 🚀

**Review by NidhiKJha — APPROVED** [review]

**Pull request merged**